src/appsettings.h

```
@@ -75,6 +80,7 @@ public slots:
     void setDefaultBuckeOffset(qint32 defaultBuckeOffset);
     void setNormalizationDivisor(qint32 normalizationDivisor);
     void setRiskScoreClasses(QList<RiskScoreClass> riskScoreClasses);
+    void setRegTokenRecieved(const QDate &recievedDate = QDate(-1, 1, 1));
```

It looks like you're using `QDate(-1, 1, 1)` to mean "invalid" here. It's cleaner and easier to set the default to `QDate()` which defaults to an invalid state. You can then test validity using `QDate::isValid()`.